internal/buffer/settings.go

```diff
@@ -49,6 +50,10 @@ func (b *Buffer) SetOptionNative(option string, nativeValue interface{}) error {
 			b.UpdateRules()
 		}
 	} else if option == "encoding" {
+		_, err := htmlindex.Get(b.Settings["encoding"].(string))
+		if err != nil {
+			b.Settings["encoding"] = "utf-8"
```

Comment:
I believe we should just return error in this case, not override previously set (valid) encoding.

Reply:
So why not just return `err` in this case?